src/shared/install.c

```diff
@@ -1492,7 +1492,7 @@ static int install_info_symlink_alias(
                 rp = skip_root(paths, i->path);
 
                 q = create_symlink(rp ?: i->path, alias_path, force, changes, n_changes);
-                if (r == 0)
+                if (q < 0)
                         r = q;
```

Comment:
yeah, it probably should indeed say: `if (r == 0 || (r > 0 && q < 0)) r = q;`

i.e. as long as r is 0 the result should always be assigned. If r is > 0 it should only be assigned if failure.

This means: the whole funciton returns 0 if nothing happened, 1 if something happened and the first error if an error happened.

Does that make sense?